src/EFCore.PG/Query/ExpressionTranslators/Internal/NpgsqlRangeTranslator.cs

```diff
@@ -37,53 +40,140 @@ namespace Npgsql.EntityFrameworkCore.PostgreSQL.Query.ExpressionTranslators.Inte
+                return null;
+
+            switch (e.Member.Name)
+            {
+            case nameof(NpgsqlRange<int>.LowerBound) when VersionAtLeast(9, 2) && !e.Type.IsNullableType():
+                return
+                    new SqlFunctionExpression(
+                        "COALESCE",
+                        e.Type,
+                        new Expression[]
+                        {
+                            new SqlFunctionExpression("lower", e.Type, new[] { e.Expression }),
+                            Expression.Default(e.Type)
+                        });
+
+            case nameof(NpgsqlRange<int>.LowerBound) when VersionAtLeast(9, 2):
+                return new SqlFunctionExpression("lower", e.Type, new[] { e.Expression });
+
+            case nameof(NpgsqlRange<int>.UpperBound) when VersionAtLeast(9, 2) && !e.Type.IsNullableType():
+                return
+                    new SqlFunctionExpression(
+                        "COALESCE",
+                        e.Type,
+                        new Expression[]
+                        {
+                            new SqlFunctionExpression("upper", e.Type, new[] { e.Expression }),
+                            Expression.Default(e.Type)
+                        });
+
+            case nameof(NpgsqlRange<int>.LowerBound) when VersionAtLeast(9, 2):
+                return new SqlFunctionExpression("upper", e.Type, new[] { e.Expression });
+
+            case nameof(NpgsqlRange<int>.IsEmpty) when VersionAtLeast(9, 2):
+                return new SqlFunctionExpression("isempty", e.Type, new[] { e.Expression });
+
+            case nameof(NpgsqlRange<int>.LowerBoundIsInclusive) when VersionAtLeast(9, 2):
+                return new SqlFunctionExpression("lower_inc", e.Type, new[] { e.Expression });
+
+            case nameof(NpgsqlRange<int>.UpperBoundIsInclusive) when VersionAtLeast(9, 2):
+                return new SqlFunctionExpression("upper_inc", e.Type, new[] { e.Expression });
+
+            case nameof(NpgsqlRange<int>.LowerBoundInfinite) when VersionAtLeast(9, 2):
+                return new SqlFunctionExpression("lower_inf", e.Type, new[] { e.Expression });
+
+            case nameof(NpgsqlRange<int>.UpperBoundInfinite) when VersionAtLeast(9, 2):
+                return new SqlFunctionExpression("upper_inf", e.Type, new[] { e.Expression });
 
             default:
                 return null;
             }
         }
+
+        #region Helpers
+
+        /// <summary>
+        /// True if <see cref="_postgresVersion"/> is null, greater than, or equal to the specified version.
+        /// </summary>
+        /// <param name="major">The major version.</param>
+        /// <param name="minor">The minor version.</param>
+        /// <returns>
+        /// True if <see cref="_postgresVersion"/> is null, greater than, or equal to the specified version.
+        /// </returns>
+        bool VersionAtLeast(int major, int minor)
```

Comment:
I totally agree :)

Reply:
Fair fair... 

Actually, I was thinking about pulling this out into an internal extension class (maybe into `Check`?) since it's used in a couple of different files now. What do you two think?

Reply:
I think it's OK for something this small to be duplicated...